### PR TITLE
test/*/run: improve signal handling in test runners

### PR DESCRIPTION
**Pull request by nyh**
When a user runs a script and presses control-C, a SIGINT (signal 2)
gets sent to every process in the script's "process group". By default,
every subprocess started by a script joins the parent's process group.

Our test/*/run test-runner scripts typically start two processes: scylla
and pytest. If we keep them in the same process group, a control-C
would kill them in a random order and that is ugly - if Scylla is
killed before pytest, we'll see a few test failures before pytest
is finally killed. So the existing code put Scylla in its own process
group, and killed it on exit after killing pytest.

But there were a few inconsistencies in our implementation, leading
to some annoying behaviors:

1. Doing "kill -2" to the runner's process (not a control-C which sends
   a signal to the process group) caused scylla and pytest to be killed
   on exit. So far so good. But, we should kill their entire process
   groups, not just the one process. This is important when pytest starts
   its own subprocesses (as happens in cql-pytest/test_tools.py),
   otherwise they just remain running.
   We need to call pgkill() instead of kill(), but also we forgot
   to start a new process group for the pytest run - so this patch
   fixes it.

2. Our exit handler - which kills the subprocesses - only gets called
   on signals which Python catches, and this is only SIGINT. Killing
   the test runner with SIGTERM or SIGHUP before this patch caused
   the subprocesses to be left running. In this patch we also catch
   SIGTERM and SIGHUP, so our exit handler is also run in that case.

Signed-off-by: Nadav Har'El <nyh@scylladb.com>

### CODE REVIEW

File: test/cql-pytest/run.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def run_with_temporary_dir(run_cmd_generator):
 def abort_run_with_temporary_dir(pid):
     tmpdir = pid_to_dir(pid)
     try:
-        os.kill(pid, 9)
+        os.killpg(pid, 9)
```
Comment (avikivity) [inline test/cql-pytest/run.py:84]: unrelated: it's bad practice to use signal numbers since they can vary across architectures. Better use symbolic names.

Comment (nyh) [reply test/cql-pytest/run.py:84]: You're right that it's bad practice but for a different reasons - readers might not remember what this "9" is.
About changing numbers, that can't realistically happen. These numbers have not changed in 50 years ;-)

Comment (avikivity) [reply test/cql-pytest/run.py:84]: They vary across architectures (not 9 though):

```

       Signal        x86/ARM     Alpha/   MIPS   PARISC   Notes
                   most others   SPARC
       ─────────────────────────────────────────────────────────────────
       SIGHUP           1           1       1       1
       SIGINT           2           2       2       2
       SIGQUIT          3           3       3       3
       SIGILL           4           4       4       4
       SIGTRAP          5           5       5       5
       SIGABRT          6           6       6       6
       SIGIOT           6           6       6       6
       SIGBUS           7          10      10      10
       SIGEMT           -           7       7      -
       SIGFPE           8           8       8       8
       SIGKILL          9           9       9       9
       SIGUSR1         10          30      16      16
       SIGSEGV         11          11      11      11
       SIGUSR2         12          31      17      17
       SIGPIPE         13          13      13      13
       SIGALRM         14          14      14      14
       SIGTERM         15          15      15      15
       SIGSTKFLT       16          -       -        7
       SIGCHLD         17          20      18      18
       SIGCLD           -          -       18      -
       SIGCONT         18          19      25      26
       SIGSTOP         19          17      23      24
       SIGTSTP         20          18      24      25
       SIGTTIN         21          21      26      27
       SIGTTOU         22          22      27      28
       SIGURG          23          16      21      29
       SIGXCPU         24          24      30      12
       SIGXFSZ         25          25      31      30
       SIGVTALRM       26          26      28      20
       SIGPROF         27          27      29      21
       SIGWINCH        28          28      20      23
       SIGIO           29          23      22      22
       SIGPOLL                                            Same as SIGIO
       SIGPWR          30         29/-     19      19
       SIGINFO          -         29/-     -       -
       SIGLOST          -         -/29     -       -
       SIGSYS          31          12      12      31
       SIGUNUSED       31          -       -       31
```

Comment (nyh) [reply test/cql-pytest/run.py:84]: The numbers 1, 2, 3, 9, and 15, have never changed on any Unix-like system for the last 50 years.
But you are right that using the name is nicer.

Comment (nyh) [reply test/cql-pytest/run.py:84]: Turns out the numbers 0, 1, 2, 3, 6, 9, 14 and 15 were standardized and officially can't change: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/kill.html
I'm surprised 11 wasn't (but people don't normally kill() with that signal anyway).

Comment (avikivity) [reply test/cql-pytest/run.py:84]: I thought 11 was standardized by Spinal Tap.

**Pull request closed without merging**